lib/Sema/TypeCheckType.cpp

```diff
@@ -1983,11 +2033,11 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
         attrs.clearAttribute(i);
       }
     }
-  } 
-
-  // If we didn't build the type differently above, build it normally now.
-  if (!ty) ty = resolveType(repr, options);
-  if (!ty || ty->is<ErrorType>()) return ty;
+  } else if (hasFunctionAttr && fnRepr) {
+    for (auto i : FunctionAttrs)
```

Comment:
It was there before, I think it's just clearing the attributes so that we don't diagnose about them later.